test/project_types/extension/models/development_server_test.rb

```diff
@@ -5,27 +5,69 @@ module Models
     class DevelopmentServerTest < MiniTest::Test
       def setup
         ShopifyCli::ProjectType.load_type(:extension)
+        @development_server = Models::DevelopmentServer.new(executable: "fake")
         super
       end
 
       def test_default_executable_is_shopify_extensions
         development_server = Models::DevelopmentServer.new
-        assert_includes(development_server.executable, "ext/shopify-cli/shopify-extensions")
+        assert_includes(development_server.executable, "ext/shopify-extensions")
       end
 
       def test_create_calls_executable_and_is_successful
         assert_nothing_raised do
-          development_server = Models::DevelopmentServer.new(executable: "fake")
           server_config = Models::ServerConfig::Root.new(extensions: [extension])
 
           CLI::Kit::System.expects(:capture3)
-            .with(development_server.executable, "create", "-", stdin_data: server_config.to_yaml)
+            .with(@development_server.executable, "create", "-", stdin_data: server_config.to_yaml)
             .returns("", nil, true)
 
-          development_server.create(server_config)
+          @development_server.create(server_config)
         end
       end
 
+      def test_build_calls_executable_and_is_successful
+        assert_nothing_raised do
+          server_config = Models::ServerConfig::Root.new(extensions: [extension])
+
+          CLI::Kit::System.expects(:capture3)
+            .with(@development_server.executable, "build", "-", stdin_data: server_config.to_yaml)
+            .returns(["", nil, mock(success?: true)])
+
+          @development_server.build(server_config)
+        end
+      end
+
+      def test_build_raises_error_on_failure
+        error_message = "an error from stdout"
+
+        expected_error = assert_raises Models::DevelopmentServer::DevelopmentServerError do
+          server_config = Models::ServerConfig::Root.new(extensions: [extension])
+
+          CLI::Kit::System.expects(:capture3)
+            .with(@development_server.executable, "build", "-", stdin_data: server_config.to_yaml)
+            .returns(["", error_message, mock(success?: false)])
+
+          @development_server.build(server_config)
+        end
+        assert_equal(expected_error.message, error_message)
+      end
+
+      # def test_serve_runs_successfully
```

Comment:
tried to test the new server functionality, but I am getting some odd errors, like `stubbing non-existent method 'fake'.append`. Could use a pair to make sure I am testing this correctly.